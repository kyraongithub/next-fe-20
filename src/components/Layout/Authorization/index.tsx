import { getCookie } from 'cookies-next';
import { useRouter } from 'next/router';
import { useEffect } from 'react';
import { AuthProps } from './Auth.interface';

const Authorization = ({ children }: AuthProps) => {
  const router = useRouter();

  useEffect(() => {
    const token = getCookie('token');
    if (!token) router.push('/login');
  }, []);

  return children;
};

export default Authorization;
