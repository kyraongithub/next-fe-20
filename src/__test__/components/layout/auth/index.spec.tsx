import Authorization from '@/components/Layout/Authorization';
import { render } from '@testing-library/react';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe('Authorization Component', () => {
  beforeEach(async () => {
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');

    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    }));
  });
  it('should render authorization', () => {
    const { container } = render(
      <Authorization>
        <h1>test</h1>
      </Authorization>
    );
    expect(container).toMatchSnapshot();
  });
});
