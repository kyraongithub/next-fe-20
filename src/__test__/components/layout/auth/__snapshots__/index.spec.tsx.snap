// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization Component should render authorization 1`] = `
<div>
  <h1>
    test
  </h1>
</div>
`;
